app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive: list["Animal"] = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name: str = name
+        self.health: int = health
+        self.hidden: bool = False
+        Animal.alive.append(self)
+
+    def check_health(self) -> None:
+        if self.health <= 0:
+            Animal.alive.remove(self)
```

Comment:
Consider checking if the animal is in the `Animal.alive` list before attempting to remove it. This will prevent potential errors if `check_health` is called multiple times or if the animal was never added to the list.